system/jlib/jlog.hpp

```diff
@@ -1342,4 +1342,183 @@ interface IComponentLogFileCreator : extends IInterface
 extern jlib_decl IComponentLogFileCreator * createComponentLogFileCreator(IPropertyTree * _properties, const char *_component);
 extern jlib_decl IComponentLogFileCreator * createComponentLogFileCreator(const char *_logDir, const char *_component);
 extern jlib_decl IComponentLogFileCreator * createComponentLogFileCreator(const char *_component);
+
+enum AccessLogRangeType
+{
+    ACRTUnknown = -1,
+    ACRTAllAvailable,
+    ACRTFirstPage,
+    ACRTirstNRows,
+    ACRTLastNYears,
+    ACRTLastNMonths,
+    ACRTLastNWeeks,
+    ACRTLastNDays,
+    ACRTLastNHours,
+    ACRTLastNMinutes,
+    ACRTLastPage,
+    ACRTGoToPage,
+    ACRTLastNRows,
+    ACRTTimeRange,
+};
+
+struct LogAccessRange
+{
+    AccessLogRangeType rangeType;
+    StringBuffer startDate;
+    StringBuffer endDate;
+    unsigned quantity;
+};
+
+enum AccessLogFilterType
+{
+    ALF_ALLAvailable = 0,
+    ALF_Component = 1,
+    ALF_WorkUnit = 2,
+    ALF_Audience = 3,
+    ALF_Class = 4,
+    ALF_NativeQuery = 5
+};
+
+struct LogAccessFilter
+{
+    AccessLogFilterType filterType;
+    StringBuffer workUnit;
+    StringArray targetAudience;
+    StringArray messageClass;
+    StringBuffer componentName;
+};
+
+struct LogAccessConditions
+{
+    LogAccessFilter filter;
+    LogAccessRange range;
+    StringArray logFieldNames;
+
+    void operator = (const LogAccessConditions & l)
+    {
+        if (l.logFieldNames.ordinality())
+        {
+            ForEachItemIn(i, l.logFieldNames)
+            {
+                logFieldNames.append(l.logFieldNames.item(i));
+            }
+        }
+    }
+
+    ~LogAccessConditions()
+    {
+        logFieldNames.kill(true);
+    }
+
+    void appendLogFieldName(const char * fieldname) {logFieldNames.append(fieldname);}
```

Comment:
trivial: spacing unusual (normally space before and after curly brackets

should this be checking that fieldname not duplicated (perhaps in _DEBUG only)?